### ISSUE TITLE
Moving spam message to submissions results in a server error and looses Stripe data

### ISSUE THREAD

**Issue by terryupton** [solspace/craft3-freeform#666]
**Describe the bug or issue you're experiencing**
So we have a registration form with Stripe integration and occasionally the odd submission gets pulled out as spam (see my  FR here: https://github.com/solspace/craft-freeform/discussions/663)

When setting this submission as 'Allow', we hit a Internal Server error.
Upon turning on debug, this is what we see.

<img width="1928" alt="Screenshot 2023-04-26 at 13 23 32" src="https://user-images.githubusercontent.com/5698596/234578466-515c8714-3ddd-4b93-878f-77f21966d5a4.png">

```
SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'errorMessage' at row 1
The SQL being executed was: INSERT INTO `craft_freeform_payments_payments` (`integrationId`, `resourceId`, `submissionId`, `subscriptionId`, `amount`, `currency`, `last4`, `status`, `metadata`, `errorCode`, `errorMessage`, `dateCreated`, `dateUpdated`, `uid`) VALUES (1, NULL, 70662, NULL, NULL, NULL, NULL, 'failed', NULL, '0', 'Unrecognized request URL (GET: /v1/payment_intents/). If you are trying to list objects, remove the trailing slash. If you are trying to retrieve an object, make sure you passed a valid (non-empty) identifier in your code. Please see https://stripe.com/docs or we can help at https://support.stripe.com/.', '2023-04-26 12:23:22', '2023-04-26 12:23:22', 'd1112e30-5905-4904-a8eb-46bf8476ac75')
Error Info: Array
(
    [0] => 22001
    [1] => 1406
    [2] => Data too long for column 'errorMessage' at row 1
)
↵
Caused by: PDOException
SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'errorMessage' at row 1
in /home/hatchendtuition/public_html/vendor/yiisoft/yii2/db/Command.php at line 1302
```

However, besides the client getting flustered at this message, the main issue, is that Stripe information has gone missing and no longer accessible via the Submission.

<img width="1470" alt="Screenshot 2023-04-26 at 13 44 58" src="https://user-images.githubusercontent.com/5698596/234579200-2701111e-00dd-42b0-9a02-f4e142d9888a.png">

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 3.8.7
- Freeform Version: 3.13.28
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Comment by kjmartens** [comment]
Sorry for the trouble and the delay @terryupton.

This might be a limitation with the Stripe integration, but I will have a developer check into this next week and get back to you. 🙂

**Comment by kjmartens** [comment]
Hi @terryupton,

There's essentially no way to have this work (and process the payment later on), as processing and success needs to happen immediately upon submit of the form. What we'll do is prevent Freeform from attempting to process the submission/payment again when clicking "Allow" button from spam folder. A future major version of Freeform will include the ability to disable spam features per form, so that part of the FR will happen. 🙂

**Comment by terryupton** [comment]
@kjmartens - as far as I am aware, the payment is taken correctly at the time but the submission then ends up in spam and then you get the server error when moving it. I was more concerned about the server error, as it concerns clients when the see this. I believe the payments are completed correctly, so I was hoping it would be a case of just linking these together from the records.

I believe that the new. spam prevention disable feature should hopefully resolve all these issues anyway.

P.S. you're up early :-)

**Comment by kjmartens** [comment]
Ah ok. Thanks for clarifying. We'll take a closer look at this and see if it's possible to retain the Stripe payment info in this case (as well as prevent the "Allow" process from attempting to process the payment again). 🙂

**Comment by kjmartens** [comment]
This will be fixed shortly in Freeform 4.0.24. 🙂 

Any chance you're upgrading to Craft 4 in the near future?

**Comment by terryupton** [comment]
@kjmartens will this fix not come to Craft 3?
